src/polygon/pickle-jar-univ3.sol

```diff
@@ -264,6 +224,155 @@ contract PickleJarUniV3Poly is ERC20, ReentrancyGuard {
         return totalLiquidity().mul(1e18).div(totalSupply());
     }
 
+    function _deposit(uint256 token0Amount, uint256 token1Amount) internal {
+        if (
+            !(token0.balanceOf(address(this)) > token0Amount) &&
+            (token0Amount != 0)
+        ) token0.safeTransferFrom(msg.sender, address(this), token0Amount);
+
+        if (
+            !(token1.balanceOf(address(this)) > token1Amount) &&
+            (token1Amount != 0)
+        ) token1.safeTransferFrom(msg.sender, address(this), token1Amount);
+
+        _balanceProportion(getLowerTick(), getUpperTick());
+    }
+
+    function _convertMatic(uint256 token0Amount, uint256 token1Amount)
+        internal
+        returns (uint256, uint256)
+    {
+        uint256 _matic = address(this).balance;
```

Comment:
Probably more appropriate to use `msg.balance` here instead, so the MATIC entitlement maps exactly to what is sent in.